src/ui/public/vislib/lib/axis.js

```diff
@@ -1,695 +1,227 @@
 import d3 from 'd3';
-import $ from 'jquery';
 import _ from 'lodash';
-import moment from 'moment';
-import VislibLibErrorHandlerProvider from 'ui/vislib/lib/_error_handler';
-import errors from 'ui/errors';
+import $ from 'jquery';
+import ErrorHandlerProvider from 'ui/vislib/lib/_error_handler';
+import AxisTitleProvider from 'ui/vislib/lib/axis_title';
+import AxisLabelsProvider from 'ui/vislib/lib/axis_labels';
+import AxisScaleProvider from 'ui/vislib/lib/axis_scale';
 
 export default function AxisFactory(Private) {
-
-  const ErrorHandler = Private(VislibLibErrorHandlerProvider);
-
+  const ErrorHandler = Private(ErrorHandlerProvider);
+  const AxisTitle = Private(AxisTitleProvider);
+  const AxisLabels = Private(AxisLabelsProvider);
+  const AxisScale = Private(AxisScaleProvider);
+  const defaults = {
+    show: true,
+    type: 'value',
+    elSelector: '.axis-wrapper-{pos} .axis-div',
+    position: 'left',
+    axisFormatter: null, // TODO: create default axis formatter
+    scale: 'linear',
+    expandLastBucket: true, //TODO: rename ... bucket has nothing to do with vis
+    inverted: false,
+    style: {
+      color: '#ddd',
+      lineWidth: '1px',
+      opacity: 1,
+      tickColor: '#ddd',
+      tickWidth: '1px',
+      tickLength: '6px'
+    }
+  };
+
+  const categoryDefaults = {
+    type: 'category',
+    position: 'bottom',
+    labels: {
+      rotate: 0,
+      rotateAnchor: 'end',
+      filter: true
+    }
+  };
   /**
-   * Adds an x axis to the visualization
+   * Appends y axis to the visualization
    *
    * @class Axis
    * @constructor
-   * @param args {{el: (HTMLElement), xValues: (Array), ordered: (Object|*),
-   * xAxisFormatter: (Function), _attr: (Object|*)}}
+   * @param args {{el: (HTMLElement), yMax: (Number), _attr: (Object|*)}}
    */
   class Axis extends ErrorHandler {
     constructor(args) {
       super();
-      this.el = args.el;
-      this.xValues = args.xValues;
-      this.ordered = args.ordered;
-      this.axisFormatter = args.type === 'category' ? args.xAxisFormatter : args.yAxisFormatter;
-      this.expandLastBucket = args.expandLastBucket == null ? true : args.expandLastBucket;
-      this._attr = _.defaults(args._attr || {});
-      this.scale = null;
-      this.domain = [args.yMin, args.yMax];
-      this.elSelector = args.type === 'category' ? '.x-axis-div' : '.y-axis-div';
-      this.type = args.type;
+      if (args.type === 'category') {
+        _.extend(this, defaults, categoryDefaults, args);
+      } else {
+        _.extend(this, defaults, args);
+      }
+
+      this._attr = args.vis._attr;
+      this.elSelector = this.elSelector.replace('{pos}', this.position);
+      this.scale = new AxisScale(this, {scale: this.scale});
+      this.axisTitle = new AxisTitle(this, this.axisTitle);
```

Comment:
What is `this.axisTilte` supposed to be equal to when it's passed in? I don't think it's equal to anything